### ISSUE TITLE
linux/debian: no 3D view after system updates

### ISSUE THREAD

**Issue by hg42** [daid/LegacyCura#749]
just want to inform you, that updating some packages made cura fail to display 3D view, making it useless.

I had this some weeks before and fixed it by reverting some updates. I don't exactly remember which, I think it was python-opengl.

But, reverting the python packages used by Cura didn't help.

I'm really tired now, so I'll continue to investigate tomorrow.

**Comment by hg42** [comment]
I checked the package history and I think the attached part should include the changes.
I'm not sure when exactly I printed my last file, so I also included updates of 2014-02-26, but I assume the problem starts on 2014-02-27.

----- snip -----

Start-Date: 2014-02-26  01:26:32
Install: liboce-foundation8:amd64 (0.15-3, automatic), liboce-visualization8:amd64 (0.15-3, automatic), liboce-ocaf8:amd64 (0.15-3, automatic), libfreeimage3:amd64 (3.15.4-3, automatic), liboce-modeling8:amd64 (0.15-3, automatic), liboce-ocaf-lite8:amd64 (0.15-3, automatic), libfm4:amd64 (1.2.0-1, automatic), libraw9:amd64 (0.15.4-1, automatic)
Upgrade: libavformat54:amd64 (9.11-1, 9.11-2), ed:amd64 (1.9-2, 1.10-1), strongswan-ike:amd64 (5.1.1-1, 5.1.1-2), libfontembed1:amd64 (1.0.45-1, 1.0.46-1), libavresample1:amd64 (9.11-1, 9.11-2), python-crypto:amd64 (2.6.1-3, 2.6.1-3+b1), librasqal3:amd64 (0.9.31-3, 0.9.32-1), libpython2.7-stdlib:amd64 (2.7.6-5, 2.7.6-6), multiarch-support:amd64 (2.18-1, 2.18-3), cups-core-drivers:amd64 (1.7.1-5, 1.7.1-6), cups-server-common:amd64 (1.7.1-5, 1.7.1-6), rpm2cpio:amd64 (4.11.1-3, 4.11.2-1), libgirepository-1.0-1:amd64 (1.36.0-2+b1, 1.38.0-2), libfm-gtk-data:amd64 (1.1.2.2-1, 1.2.0-1), libc6-dev-x32:amd64 (2.18-1, 2.18-3), libavcodec-dev:amd64 (9.11-1, 9.11-2), golang:amd64 (1.2-2, 1.2-3), libavcodec-extra-54:amd64 (9.11-1, 9.11-2), libstrongswan:amd64 (5.1.1-1, 5.1.1-2), libc6-dev-i386:amd64 (2.18-1, 2.18-3), automake:amd64 (1.14.1-2, 1.14.1-3), wireless-regdb:amd64 (2013.02.13-1, 2013.11.27-1), libfontconfig1:amd64 (2.11.0-4, 2.11.0-5), libfontconfig1:i386 (2.11.0-4, 2.11.0-5), libcups2:amd64 (1.7.1-5, 1.7.1-6), ack-grep:amd64 (2.12-1, 2.12-2), cups-daemon:amd64 (1.7.1-5, 1.7.1-6), golang-go-linux-amd64:amd64 (1.2-2, 1.2-3), libcupsmime1:amd64 (1.7.1-5, 1.7.1-6), cups-client:amd64 (1.7.1-5, 1.7.1-6), libc-dev-bin:amd64 (2.18-1, 2.18-3), libcupsimage2:amd64 (1.7.1-5, 1.7.1-6), nfs-kernel-server:amd64 (1.2.8-5, 1.2.8-6), fontconfig:amd64 (2.11.0-4, 2.11.0-5), libavformat-dev:amd64 (9.11-1, 9.11-2), libc-bin:amd64 (2.18-1, 2.18-3), libc6:amd64 (2.18-1, 2.18-3), libc6:i386 (2.18-1, 2.18-3), python2.7:amd64 (2.7.6-5, 2.7.6-6), cups:amd64 (1.7.1-5, 1.7.1-6), nfs-common:amd64 (1.2.8-5, 1.2.8-6), ardour3:amd64 (3.5.308~dfsg-1, 3.5.357~dfsg-1), fontconfig-config:amd64 (2.11.0-4, 2.11.0-5), libswscale-dev:amd64 (9.11-1, 9.11-2), rpm-common:amd64 (4.11.1-3, 4.11.2-1), libswscale2:amd64 (9.11-1, 9.11-2), strongswan-nm:amd64 (5.1.1-1, 5.1.1-2), eject:amd64 (2.1.5+deb1+cvs20081104-13, 2.1.5+deb1+cvs20081104-13.1), firmware-ralink:amd64 (0.40, 0.41), glibc-doc:amd64 (2.18-1, 2.18-3), libavutil-dev:amd64 (9.11-1, 9.11-2), libmtp9:amd64 (1.1.6-20-g1b9f164-1, 1.1.6-20-g1b9f164-2), gir1.2-freedesktop:amd64 (1.36.0-2+b1, 1.38.0-2), libpython2.7-dev:amd64 (2.7.6-5, 2.7.6-6), cups-filters-core-drivers:amd64 (1.0.45-1, 1.0.46-1), libfm-data:amd64 (1.1.2.2-1, 1.2.0-1), freecad:amd64 (0.13.2935-dfsg-1.1, 0.13.2935-dfsg-1.1+b1), dictionaries-common:amd64 (1.22.0, 1.22.1), gir1.2-glib-2.0:amd64 (1.36.0-2+b1, 1.38.0-2), libpython2.7:amd64 (2.7.6-5, 2.7.6-6), rpm:amd64 (4.11.1-3, 4.11.2-1), libcupscgi1:amd64 (1.7.1-5, 1.7.1-6), librpm3:amd64 (4.11.1-3, 4.11.2-1), libdlrestrictions1:amd64 (0.15.12, 0.15.13), cups-filters:amd64 (1.0.45-1, 1.0.46-1), amsynth:amd64 (1.3.2-2, 1.4.1-2), system-config-printer:amd64 (1.4.3-1, 1.4.3-4), python2.7-dev:amd64 (2.7.6-5, 2.7.6-6), libcupsppdc1:amd64 (1.7.1-5, 1.7.1-6), firmware-realtek:amd64 (0.40, 0.41), libmtp-common:amd64 (1.1.6-20-g1b9f164-1, 1.1.6-20-g1b9f164-2), debugedit:amd64 (4.11.1-3, 4.11.2-1), python2.7-minimal:amd64 (2.7.6-5, 2.7.6-6), golang-doc:amd64 (1.2-2, 1.2-3), libc6-i386:amd64 (2.18-1, 2.18-3), cups-common:amd64 (1.7.1-5, 1.7.1-6), init-system-helpers:amd64 (1.17, 1.18), libavdevice53:amd64 (9.11-1, 9.11-2), librpmsign1:amd64 (4.11.1-3, 4.11.2-1), python-dulwich:amd64 (0.9.5-1, 0.9.5-2), system-config-printer-udev:amd64 (1.4.3-1, 1.4.3-4), libavcodec-extra:amd64 (9.10-1, 9.11-2), python-cupshelpers:amd64 (1.4.3-1, 1.4.3-4), golang-src:amd64 (1.2-2, 1.2-3), cups-ppdc:amd64 (1.7.1-5, 1.7.1-6), libmtp-runtime:amd64 (1.1.6-20-g1b9f164-1, 1.1.6-20-g1b9f164-2), librpmbuild3:amd64 (4.11.1-3, 4.11.2-1), libc6-dbg:amd64 (2.18-1, 2.18-3), golang-go:amd64 (1.2-2, 1.2-3), libfontconfig1-dev:amd64 (2.11.0-4, 2.11.0-5), librpmio3:amd64 (4.11.1-3, 4.11.2-1), locales:amd64 (2.18-1, 2.18-3), libpython2.7-minimal:amd64 (2.7.6-5, 2.7.6-6), libavutil52:amd64 (9.11-1, 9.11-2), libc6-dev:amd64 (2.18-1, 2.18-3), ipython:amd64 (1.2.0~rc1-1, 1.2.1-1), libcupsfilters1:amd64 (1.0.45-1, 1.0.46-1), libtbb2:amd64 (4.2~20130725-1.1, 4.2~20130725-1.2), libc6-x32:amd64 (2.18-1, 2.18-3)
Remove: liboce-foundation6:amd64 (0.12-4), liboce-visualization6:amd64 (0.12-4), liboce-ocaf6:amd64 (0.12-4), liboce-modeling6:amd64 (0.12-4), liboce-ocaf-lite6:amd64 (0.12-4)
End-Date: 2014-02-26  01:28:38

Start-Date: 2014-02-27  10:14:27
Install: libfm-gtk4:amd64 (1.2.0-1, automatic)
Upgrade: ed:amd64 (1.10-1, 1.10-2), tap-plugins:amd64 (0.7.2-1, 0.7.2-2), python-dbus:amd64 (1.2.0-2+b1, 1.2.0-2+b2), apt:amd64 (0.9.15.4, 0.9.15.4+b1), libfam0:amd64 (2.7.0-17, 2.7.0-17.1), python-crypto:amd64 (2.6.1-3+b1, 2.6.1-4), libpython2.7-stdlib:amd64 (2.7.6-6, 2.7.6-7), librxtx-java:amd64 (2.2pre2-11, 2.2pre2-12), libaprutil1:amd64 (1.5.3-1, 1.5.3-1+b1), dbus:amd64 (1.8.0-1, 1.8.0-2), python-qt4-dbus:amd64 (4.10.3+dfsg1-1, 4.10.3+dfsg1-1+b1), smartmontools:amd64 (6.2+svn3841-1.1, 6.2+svn3841-1.2), ruby-build:amd64 (20130227-1, 20140224.1-1), python-numpy:amd64 (1.7.1-3, 1.7.1-3+b1), python-qt4:amd64 (4.10.3+dfsg1-1, 4.10.3+dfsg1-1+b1), python-qt4-gl:amd64 (4.10.3+dfsg1-1, 4.10.3+dfsg1-1+b1), python3-gi-cairo:amd64 (3.10.2-2, 3.10.2-2+b1), apt-utils:amd64 (0.9.15.4, 0.9.15.4+b1), python3-pyqt4:amd64 (4.10.3+dfsg1-1, 4.10.3+dfsg1-1+b1), rubberband-vamp:amd64 (1.8.1-4, 1.8.1-5), zita-at1:amd64 (0.2.3-4, 0.2.4-1), qtcreator-doc:amd64 (2.8.1-3, 3.0.1+dfsg-1), netdiscover:amd64 (0.3beta6+20080409-5, 0.3beta6+20080409-5.1), gparted:amd64 (0.17.0-5, 0.18.0-1), python2.7:amd64 (2.7.6-6, 2.7.6-7), libapt-inst1.5:amd64 (0.9.15.4, 0.9.15.4+b1), python3-dbus.mainloop.qt:amd64 (4.10.3+dfsg1-1, 4.10.3+dfsg1-1+b1), libdbus-1-3:amd64 (1.8.0-1, 1.8.0-2), libdbus-1-3:i386 (1.8.0-1, 1.8.0-2), libyaml-tiny-perl:amd64 (1.60-1, 1.61-1), rabbitvcs-cli:amd64 (0.15.2-1, 0.15.3-1), invada-studio-plugins-lv2:amd64 (1.2.0+repack0-4, 1.2.0+repack0-5), rubberband-ladspa:amd64 (1.8.1-4, 1.8.1-5), python-gi:amd64 (3.10.2-2, 3.10.2-2+b1), libpython2.7-dev:amd64 (2.7.6-6, 2.7.6-7), dictionaries-common:amd64 (1.22.1, 1.22.2), libapt-pkg4.12:amd64 (0.9.15.4, 0.9.15.4+b1), libdbus-1-dev:amd64 (1.8.0-1, 1.8.0-2), libpython2.7:amd64 (2.7.6-6, 2.7.6-7), python-sip:amd64 (4.15.4-1, 4.15.4-1+b1), python-gi-cairo:amd64 (3.10.2-2, 3.10.2-2+b1), libcloog-isl4:amd64 (0.18.1-3, 0.18.2-1), librubberband2:amd64 (1.8.1-4, 1.8.1-5), python3-gi:amd64 (3.10.2-2, 3.10.2-2+b1), python2.7-dev:amd64 (2.7.6-6, 2.7.6-7), jack-tools:amd64 (20101210-2, 20101210-2.1), python-zope.interface:amd64 (4.0.5-1+b1, 4.0.5-1+b2), python2.7-minimal:amd64 (2.7.6-6, 2.7.6-7), libcwiid1:amd64 (0.6.00+svn201-3+b1, 0.6.00+svn201-3.1), pcmanfm:amd64 (1.1.2-1, 1.2.0-1), dbus-x11:amd64 (1.8.0-1, 1.8.0-2), rabbitvcs-core:amd64 (0.15.2-1, 0.15.3-1), kicad-common:amd64 (0.20131208+bzr4024-1, 0.20140224+bzr4027-1), python3-sip:amd64 (4.15.4-1, 4.15.4-1+b1), python3-dbus:amd64 (1.2.0-2+b1, 1.2.0-2+b2), libpython2.7-minimal:amd64 (2.7.6-6, 2.7.6-7), ipython:amd64 (1.2.1-1, 1.2.1-2), kicad:amd64 (0.20131208+bzr4024-1, 0.20140224+bzr4027-1)
Downgrade: thunderbird:amd64 (27.0~b1+build1-0ubuntu0.13.10.1, 24.3.0+build2-0ubuntu0.13.10.1)
Remove: libfm-gtk3:amd64 (1.1.2.2-1), libfm3:amd64 (1.1.2.2-1)
End-Date: 2014-02-27  10:15:31

----- end -----

**Comment by hg42** [comment]
I already downgraded these packages yesterday which didn't help:

freeglut3:amd64/testing 2.6.0-4ubuntu1 upgradeable to 2.8.1-1
freeglut3-dev:amd64/testing 2.6.0-4ubuntu1 upgradeable to 2.8.1-1
python-numpy:amd64/testing 1:1.7.1-1ubuntu1 upgradeable to 1:1.7.1-3+b1
python-opengl:all/testing 3.0.1-1 upgradeable to 3.0.2-1

**Comment by hg42** [comment]
Ok, downgrading numpy with python-numpy_1.6.2-1.2_amd64.deb works.
I got this package from debian/wheezy.
Upgrading python-opengl to 3.0.2 again works, too.

So Cura seems to have an issue with numpy somewhere above 1.6.2 at least starting with 1.7.1-1ubuntu1.

Btw. I didn't get any error messages or error outputs on stderr/stdout.

**Comment by nallath** [comment]
I've just updated my system to the newest packages and the 3D rendering works just fine.

**Comment by hg42** [comment]
now I'm totally puzzled...
Just for fun, and to check, if it's eventually a problem of numpy 1.7.x, I upgraded to 1.8.0.
And it worked!

I thought, well let's verify the opposite, so test if 1.7.x doesn't work.

Now I have all the packages at the same versions present when I got the problem.
And it works...

Some packages like freecad may have different versions now. They were downgraded, when installing numpy 1.6.2 from an older debian distribution.
So I'll investigate further.

**Comment by hg42** [comment]
@nallath which distribution do you use?

I'm usually on debian sid, with some packages from ubuntu (currently saucy, firefox and some PPAs).

**Comment by hg42** [comment]
I checked the packages I use from ubuntu and I'm pretty sure those don't have any effect on cura (music packages, smb, languages like haxe, julia, but nothing affecting graphics or math).
I also upgraded all packages that were downgraded before (all depend on freecad) and this didn't change anything).

So I start thinking of different reasons, now.
May be CuraEngine wasn't in sync with Cura gui? Could that cause a completely empty graphics view? Meaning a gray dialog background, like in the left part of the window. In graphics mode the background outside the build area has a little bit darker gray color on my system.

**Comment by daid** [comment]
A total gray window can only happen if something goes really major wrong at an OpenGL level. So I would look at X11 stuff, or Kernel drivers.

**Comment by hg42** [comment]
installing a package via dpkg doesn't show up in apt logs, so I checked dpkg logs instead.
I extracted all package names involved in the update(s) that lead to the problem and the install that fixed the problem.

Common packages of update 2014-02-26 and installing numpy 1.6.2 are:

desktop-file-utils:amd64
freecad:amd64
gnome-menus:amd64
libc-bin:amd64
libfreeimage3:amd64
liboce-foundation8:amd64
liboce-modeling8:amd64
liboce-ocaf8:amd64
liboce-ocaf-lite8:amd64
liboce-visualization8:amd64
libraw9:amd64
man-db:amd64
menu:amd64
mime-support:all
shared-mime-info:amd64

Common packages of update 2014-02-27 and installing numpy 1.6.2 are:

desktop-file-utils:amd64
gnome-menus:amd64
libc-bin:amd64
man-db:amd64
menu:amd64
mime-support:all
python-numpy:amd64
python-support:all
shared-mime-info:amd64

If I exclude some packages, that shouldn't influence the problem, e.g. package triggers like menu and mime-support etc., I get:

freecad:amd64
libc-bin:amd64
libfreeimage3:amd64
liboce-foundation8:amd64
liboce-modeling8:amd64
liboce-ocaf8:amd64
liboce-ocaf-lite8:amd64
liboce-visualization8:amd64
libraw9:amd64
python-numpy:amd64
python-support:all

here liboce*, libfreeimage3, libraw9 are only used by freecad, so all remaining packages are these:

libc-bin:amd64
python-numpy:amd64
python-support:all

So there still seems to be some issue with numpy.

Unfortunately I don't remember exactly what I did when the problem began.
May be I first started my installed cura (compiled by me some days before).
But in any case (as second or only step) I pulled the repo, merged, compiled cura by package.sh.
Then I only tried reverting some packages, which didn't help.
Installing numpy 1.6.2 then fixed the problem, and this only changed freecad and some dependent packages.
So it seems no opengl or x11 was involved in these steps.

Anyway, I cannot recreate the problem by installing package versions back and forth.
So it may be caused by something like inconsistency in my packages.

Should I close this issue now?
I assume I may get this issue again, as I did some weeks before.
But I can reopen it then.
